presto-main/src/main/java/com/facebook/presto/server/HttpRemoteTask.java

```diff
@@ -380,6 +395,15 @@ private synchronized void updateTaskInfo(TaskInfo newValue, List<TaskSource> sou
                 pendingSourceSplitCount -= removed;
             }
         }
+
+        fireSplitChanged(getPartitionedSplitCount() - oldSplitCount);
```

Comment:
Assuming you can make the change I mentioned above, you also need to change this to be
```
if (newValue.getState.isDone()) {
  fireSplitChanged(-oldSplitCount);
}
else {
  fireSplitChanged(getPartitionedSplitCount() - oldSplitCount);
}